lib/assets/javascripts/turbolinks.js.coffee

```diff
@@ -177,12 +177,10 @@ swapNodes = (targetBody, existingNodes, options) ->
       else
         targetNode = targetNode.cloneNode(true)
         existingNode.parentNode.replaceChild(targetNode, existingNode)
-        if targetNode.nodeName == 'SCRIPT' && targetNode.getAttribute("data-turbolinks-eval") != "false"
-          executeScriptTag(targetNode)
```

Comment:
I think it was some combination of not executing `turbolinks-eval` false and `data-turbolinks-permanent` more than once. 

We should add a test for `data-turbolinks-permanent` only executing once. Also, `data-turbolinks-eval` false should probably override all.